pkg/search/proxy/store/util.go

```diff
@@ -66,10 +92,45 @@ func (m *multiClusterResourceVersion) String() string {
 	if len(m.rvs) == 0 {
 		return ""
 	}
-	buf, _ := json.Marshal(&m.rvs)
+	buf := marshalRvs(m.rvs)
 	return base64.RawURLEncoding.EncodeToString(buf)
 }
 
+func marshalRvs(rvs map[string]string) []byte {
+	// We must make sure the returned ResourceVersion string is stable, because client might use this string
+	// for equality comparison. But hashmap's key iteration order is not determined.
+	// So instead we convert the map to a slice, and the slice must be sorted by `Cluster` field
+	// before encoding.
+
+	type onWireRvs struct {
+		Cluster         string
+		ResourceVersion string
+	}
+
+	slice := make([]onWireRvs, 0, len(rvs))
+
+	for clusterName, version := range rvs {
+		obj := onWireRvs{clusterName, version}
+		slice = append(slice, obj)
+	}
+
+	sort.Slice(slice, func(i, j int) bool {
+		return slice[i].Cluster < slice[j].Cluster
+	})
+
+	var encoded []byte
```

Comment:
init this slice with a capcity? we can roughly estimate by:
```
cap := (len(slice[0].Cluster)+len(slice[0].ResourceVersion)+2)*len(slice)
```

Reply:
OK